sklego/pandas_utils.py

```diff
@@ -332,23 +347,19 @@ def _add_lagged_pandas_columns(df, cols, lags, drop_na):
 
     Returns
     -------
-    pd.DataFrame
+    DataFrame
         Dataframe with concatenated lagged cols.
     """
 
     cols = as_list(cols)
 
-    # Indexes are not supported as pandas column names may be
-    # integers themselves, introducing unexpected behaviour
-    if not all([col in df.columns.values for col in cols]):
+    if not all([col in df.columns for col in cols]):
         raise KeyError("The column does not exist")
 
-    combos = (df[col].shift(-lag).rename(col + str(lag)) for col in cols for lag in lags)
+    answer = df.with_columns(nw.col(col).shift(-lag).alias(col + str(lag)) for col in cols for lag in lags)
```

Comment:
Any plan for the `.name` namespace? Asking for a friend 😂
I can take a look myself during the weekend 😇

Reply:
definitely!